sklearn/metrics/pairwise.py

```diff
@@ -176,7 +191,7 @@ def check_pairwise_arrays(
                 "(n_queries, n_indexed). Got (%d, %d) "
                 "for %d indexed." % (X.shape[0], X.shape[1], Y.shape[0])
             )
-    elif X.shape[1] != Y.shape[1]:
+    elif X.ndim > 1 and X.shape[1] != Y.shape[1]:
```

Comment:
I think that reading the code, the validation could be a bit surprising because we are validating the length but not for the case where `check_only_length` is True ;)

Maybe the following would be better:

```suggestion
    elif not check_length_only and X.shape[1] != Y.shape[1]:
        # We ensure that X and Y are 2D
        ...
    elif check_length_only and len(X) != len(Y):
        raise ValueError(...)
```

Reply:
>     elif check_length_only and len(X) != len(Y):

Why is this check required? 
Pairwise distances for strings, Y is not supposed to be passed, and hence I think there is nothing for us to check when `check_length_only` is set to True

Reply:
ok, so if the user is only passing `X`, then Y is set to same as `X` and the condition `len(X) != len(Y)` is not possible.

But this condition would be possible if we also allow the possibility of  passing X and Y (both list of strings). 
I will add this part (and also some test cases for this scenario)